### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by timmintam**
In the `_trace_of_prod` method of the `ProductFrame` class, we had these lines :
```
# If the label does exist, we multiply the coefficient into our summand.
# The factor 2*N_qubit comes from Tr[(P_1...P_N)^2] = 2*N.
summand *= coeff * 2 * povm.num_subsystems
```
but actually the correct factor is `2^N_qubit`.

### CODE REVIEW

**Review by mrossinek — COMMENTED** [review]
Please add a release note which explains the bug.
And ideally also add a unittest which ensures that this works as expected now

File: releasenotes/notes/release_fix_trace_of_products-9bb81dfc2724ed1d.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+fixes:
+  - |
+    Fixing a bug in the ``ProductFrame._trace_of_prod`` method. 
+    This method computes the trace of the product of two operators.
+    It requires at some point the value of ``Tr[(P_N)^2]`` where ``P_N`` is an ``N``-qubit Pauli string. 
+    The correct value of this trace is ``2^N`` but was previously set to ``2*N``. 
+    This release fixes this issue.
+    Note that this issue was only impacting ``MultiQubitFrame`` for ``num_qubit>=3`` or product of such frames.
```
Comment (mrossinek) [inline releasenotes/notes/release_fix_trace_of_products-9bb81dfc2724ed1d.yaml:9]: I think it is fine if we keep it less detailed than that.
How about the following?
```suggestion
    Fixes a bug in the :meth:`.ProductFrame.analysis` method occurring for :class:`.MultiQubitFrame` instances
    with ``num_qubits>2``. This bug affected the outcome of other methods such as
    :meth:`.BasePOVM.get_prob` and :meth:`.BaseDual.get_omegas`.
```

Comment (timmintam) [reply releasenotes/notes/release_fix_trace_of_products-9bb81dfc2724ed1d.yaml:9]: Okay sounds good 👍

**Review by mrossinek — APPROVED** [review]
Nice, thanks!

**Review by mrossinek — CHANGES_REQUESTED** [review]
Actually, hold on, this test does not seem to cover the actual problem as it also passes on `main`

File: test/quantum_info/test_product_frame.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    def test_analysis(self):
+        """Test that the ``analysis`` method works correctly."""
+        paulis = ["I", "X", "Y", "Z"]
+        frame_1_qubit = MultiQubitFrame([Operator.from_label(label) for label in paulis])
+
+        num_qubit_max = 4
+
+        for num_qubit in range(1, num_qubit_max + 1):
+            frame_product = ProductFrame.from_list(frames=num_qubit * [frame_1_qubit])
```
Comment (mrossinek) [inline test/quantum_info/test_product_frame.py:32]: This is not what we actually want to test. Rather, we would want the labels inside the `frame_1_qubit` to be of different length, right?

So smth along the lines of the following:
```python
        for num_qubit in range(1, 4):
            paulis = ["I" * num_qubit, "X" * num_qubit, "Y" * num_qubit, "Z" * num_qubit]
            frame_1_qubit = MultiQubitFrame([Operator.from_label(label) for label in paulis])

            frame_product = ProductFrame.from_list(frames=[frame_1_qubit])
```

At least that fails on `main` but works in this PR

Comment (timmintam) [reply test/quantum_info/test_product_frame.py:32]: Yes the `frame_product`does not test the specific issue mentioned above but I still added it as an other check. The actual check for the issue we had is done with the frame `frame_n_qubit` a few lines below

**Review by mrossinek — APPROVED** [review]

**Pull request merged**